### PR TITLE
support etcd

### LINKED ISSUE #55: Storage Adapter Etcd
Hi,
I'm writing the storage adapter for [Etcd](/coreos/etcd). Travis doesn't have an `etcd-service` available.
How would you manage to test it in CI?

### PR DESCRIPTION
**Pull request by alphmth**
Fixed #55

### CODE REVIEW

File: packages/etcd/package.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+	"name": "@keyv/etcd",
+	"version": "1.0.0",
+	"description": "Etcd storage adapter for Keyv",
+	"main": "src/index.js",
+	"scripts": {
+		"test": "xo --fix && nyc ava",
```
Comment (jaredwray) [inline packages/etcd/package.json]: please remove the `--fix`

File: packages/etcd/src/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const EventEmitter = require('events');
+const { Etcd3 } = require('etcd3');
+
+class KeyvEtcd extends EventEmitter {
+	constructor(url, options) {
+		super();
+		this.ttlSupport = false;
```
Comment (jaredwray) [inline packages/etcd/src/index.js]: can we make this an option where you can set it too true? If ttlSupport is valid is there additional options needed to make it work?

File: packages/etcd/test/test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const test = require('ava');
+const KeyvEtcd = require('this');
+
+const etcdURL = '127.0.0.1:2379';
+
```
Comment (jaredwray) [inline packages/etcd/test/test.js:14]: Can you please add in the standard test-suite into this as it is used for all testing.

```javascript
keyvOfficialTests(test, Keyv, 'mysql://root@localhost/keyv_test', 'mysql://foo');

const store = () => new KeyvMysql('mysql://root@localhost/keyv_test');
keyvTestSuite(test, Keyv, store);
```

**Review by jaredwray — CHANGES_REQUESTED** [review]

File: packages/etcd/test/test.js (hunk 3 above)
Comment (stropicall) [reply packages/etcd/test/test.js:14]: @jaredwray Can you help me in handling errors?  The documentation says to handle errors as specified in [link](https://microsoft.github.io/etcd3/interfaces/ioptions.html#faulthandling)

But Its not working

Comment (jaredwray) [reply packages/etcd/test/test.js:14]: When you add in the code above does it fail locally? If so, have you tried to use a try/catch around the code and if it fails to handle it correctly?

**Comment by jaredwray** [comment]
@alphmth - is this ready for review? Did you get all the changes in? If so, let me know when this is ready.

File: packages/etcd/test/test.js (hunk 3 above)
Comment (stropicall) [reply packages/etcd/test/test.js:14]: I have fixed it with help of this https://github.com/microsoft/etcd3/issues/105

**Comment by alphmth** [comment]
@jaredwray namespace test is failing so working on that. hopefully by tomorrow you can review

**Review by jaredwray — APPROVED** [review]

**Pull request merged**